lib/sidekiq_unique_jobs/config.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module SidekiqUniqueJobs
+  # Shared class for dealing with gem configuration
+  #
+  # @author Mauro Berlanda <mauro.berlanda@gmail.com>
+  class Config < Concurrent::MutableStruct.new(
+    :default_lock_timeout,
+    :enabled,
+    :unique_prefix,
+    :logger,
+    :locks,
+    :strategies,
+  )
+    DEFAULT_LOCKS = {
+      until_and_while_executing: SidekiqUniqueJobs::Lock::UntilAndWhileExecuting,
+      until_executed: SidekiqUniqueJobs::Lock::UntilExecuted,
+      until_executing: SidekiqUniqueJobs::Lock::UntilExecuting,
+      until_expired: SidekiqUniqueJobs::Lock::UntilExpired,
+      until_timeout: SidekiqUniqueJobs::Lock::UntilExpired,
+      while_executing: SidekiqUniqueJobs::Lock::WhileExecuting,
+      while_executing_reject: SidekiqUniqueJobs::Lock::WhileExecutingReject,
+    }.freeze
+
+    DEFAULT_STRATEGIES = {
+      log: SidekiqUniqueJobs::OnConflict::Log,
+      raise: SidekiqUniqueJobs::OnConflict::Raise,
+      reject: SidekiqUniqueJobs::OnConflict::Reject,
+      replace: SidekiqUniqueJobs::OnConflict::Replace,
+      reschedule: SidekiqUniqueJobs::OnConflict::Reschedule,
+    }.freeze
+
+    class << self
+      def default
```

Comment:
The only thing I immediately don't like is `class << self`. I've avoided it like the plague because I find that additional duplication of self very revealing. Especially in large classes where it is difficult to discern what level of nesting I'm at so please do use `self.default`. In my opinion not all duplication are to be avoided :)

I like having the `default` here and using it from the top level though! ❤️